src/llmcompressor/modifiers/quantization/gptq/base.py

```diff
@@ -241,34 +242,76 @@ def compress_modules(self):
         """
         Quantize modules which have been calibrated
         """
-        for module in list(self._num_samples.keys()):
-            name = self._module_names[module]
-            num_samples = self._num_samples[module]
-            quant_args = getattr_chain(module, "quantization_scheme.weights")
-
-            logger.info(f"Quantizing {name} using {num_samples} samples")
-            with torch.no_grad(), align_module_device(
-                module
-            ), self._maybe_onload_hessian(module), CompressionLogger(
-                module
-            ) as comp_logger:
-                loss, quantized_weight, scale, zero_point, g_idx = quantize_weight(
-                    module=module,
-                    quant_args=quant_args,
-                    hessians_dict=self._hessians,
-                    blocksize=self.block_size,
-                    percdamp=self.dampening_frac,
+        modules_to_quantize = list(self._num_samples.keys())
+        if not modules_to_quantize:
+            return
+
+        gpu_ids = list(range(torch.cuda.device_count()))
+        if not gpu_ids:
+            logger.info("No CUDA devices found. Running GPTQ sequentially on CPU.")
+            for module in modules_to_quantize:
+                self._quantize_and_update_module(module)
+            return
+
+        logger.info(f"Starting parallel GPTQ on {len(gpu_ids)} GPUs...")
+        original_device = get_execution_device(modules_to_quantize[0])
+
+        with ThreadPoolExecutor(max_workers=len(gpu_ids)) as executor:
```

Comment:
we might be able to shorten this using a different part of the concurrent.futures API. Can try this map feature instead of needing another for loop to call `future.result()`?

```python
import concurrent.futures
import time

def task_function_simple(x):
    time.sleep(1) # Simulate work
    return x * x

def main_map():
    inputs = [1, 2, 3, 4, 5]
    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
        # map automatically submits tasks and yields results in order
        results = executor.map(task_function_simple, inputs)
        print("Tasks submitted. Results in order:")
        for result in results:
            print(f"Result: {result}")

if __name__ == "__main__":
    main_map()
```